rules/preprocessing.smk

```diff
@@ -28,49 +28,60 @@ configfile: "configs/configuration.yaml"
 
 rule aggregate:
     input:
-        sql_files=expand("data/{plate_id}.sqlite", plate_id=PLATE_IDS),
+        sql_files="data/{plate_id}.sqlite", 
         barcodes="data/barcode_platemap.csv",
         metadata="data/metadata",
     output:
-        cell_counts=expand(
-            "results/preprocessing/{plate_id}_cell_counts.tsv", plate_id=PLATE_IDS
-        ),
-        aggregate_profile=expand(
-            "results/preprocessing/{plate_id}_aggregate.csv.gz", plate_id=PLATE_IDS
-        ),
+        cell_counts="results/preprocessing/{plate_id}_cell_counts.tsv",
+        aggregate_profile="results/preprocessing/{plate_id}_aggregate.csv.gz",
     conda:
         "../envs/cytominer_env.yaml"
     params:
         aggregate_config=config["config_paths"]["single_cell"],
-    threads: config["analysis_configs"]["preprocessing"]["threads"]
-    script:
-        "../scripts/aggregate_cells.py"
+    shell:
```

Comment:
I thought that `python` was the .smk command we should be using here. Is there an advantage to `shell`?

Reply:
Hmmmm.... Good point. 

I Learned it from this [tutorial]( http://www.hpc-carpentry.org/hpc-python/12-resources/index.html) under the `Managing CPU's section` 

It seems that the `shell` provides parallelization support. I am not too sure. 

I could be wrong. I can try with the `script` rule attribute to see if it works. If it works with the `script` rule attribute, then there will be no use for the `argparse` imports since `snakemake` automatically inputs the given parameters into the script at runtime.